### PR TITLE
Point2Point2Range, Pose2Point2BearingRange, PriorPoint2 factors

### LINKED ISSUE #48: Add `Pose2Point2BearingRange` factor type
Required for ICRA tutorial 1.

### PR DESCRIPTION
**Pull request by GearsAD**
Resolves #48, #51, and #52.

Currently tests failing because of core deserialization issue: https://github.com/JuliaRobotics/RoME.jl/issues/563

### CODE REVIEW

File: tests/test_inference_types_serialization.py [hunk 1]
```diff
@@ -6,17 +6,20 @@
...
+    PriorPoint2,
     PriorPose2,
     Uniform,
 )
 from src.navability.entities.factor.inferencetypes import LinearRelative
 
 
 def test_prior():
-    pose = Prior(Z=Normal(0, 2))
+    pose = Prior(Normal(0, 2))
```
Comment (dehann) [inline tests/test_inference_types_serialization.py:22]: yeah, Julia's `Base.@kwdef` is pretty cool.  The original "direct" constructors are still there, but the full keyword constructor with optional defaults and required fields are constructed for you at compile time.

**Review by dehann — APPROVED** [review]
for posterity, doing xref:
- https://github.com/JuliaRobotics/Caesar.jl/issues/311

**Review by jim-hill-r — APPROVED** [review]

File: tests/test_inference_types_serialization.py (hunk 1 above)
Comment (GearsAD) [reply tests/test_inference_types_serialization.py:22]: Er, this is Python :D The equivalent Julia version is a dataclass in Python, it's got a few more bells and whistles but similar concept.

Comment (GearsAD) [reply tests/test_inference_types_serialization.py:22]: I'm going to take this as a compliment that the API's look almost exactly the same :tada:

**Comment by GearsAD** [comment]
I'm going to merge all these and then we can confirm everything works against main (fix issues in new issues), and when all good we can tag.

**Pull request merged**